### PR TITLE
feat: ja translate develop/datatypes

### LINKED ISSUE #184: [JP translation] data types
https://github.com/momentohq/public-dev-docs/blob/main/i18n/ja/docusaurus-plugin-content-docs/current/develop/datatypes.md

### PR DESCRIPTION
**Pull request by riywo**
Close #184

### CODE REVIEW

**Comment by riywo** [comment]
@poppoerika Check this out!

File: i18n/ja/docusaurus-plugin-content-docs/current/develop/datatypes.md [hunk 1]
```diff
@@ -1,28 +1,31 @@
...
 pagination_prev: null
 pagination_next: null
-description: Learn the core data types in Momento Cache to match up with common types in modern programming languages and information about each.
+description: Momento Cache のコアとなるデータ型を学び、現代的なプログラミング言語のに共通する型との対応をみてみましょう。
 ---
 
-# Supported data types in Momento Cache and Momento Topics
-Momento Cache provides a group of core data types to match up with common types in various modern programming languages. This page has a list of the core data structures and information about each of them.
+# Momento Cache と Momento Topics でサポートしているデータ型
+Momento Cache は、現代的な様々なプログラミング言語に共通する型と対応する、コアとなるデータ型を提供しています。このページでは、コアのデータ構造の一覧と、それぞれの情報を紹介しています。
```
Comment (poppoerika) [inline i18n/ja/docusaurus-plugin-content-docs/current/develop/datatypes.md]: Do we need `、` here?
`現代的な様々なプログラミング言語に共通する型と対応する、`

File: i18n/ja/docusaurus-plugin-content-docs/current/develop/datatypes.md [hunk 2]
```diff
@@ -1,28 +1,31 @@
...
-### Byte arrays
-All data stored is represented as byte arrays. The Momento SDKs provide easy methods you can use to store your data as raw bytes or as portable UTF-8 strings. Those can be in the form of scalar data, such as strings, characters, numbers, and binary data, such as images and Protocol Buffers serialized arrays.
+### バイト配列
 
-### Collection data types (CDTs)
-Collection Data Types are a grouping of related data into a single item. They are stored as byte array values and each has their own set of API calls in Momento SDKs. For example, DictionaryFetch, ListFetch, and SetFetch.
+保存される全てのデータはバイト配列で表現されます。Momento SDK は、生のバイトやUTF-8 の文字列として皆さんのデータを保存するために簡単に使えるメソッドを提供してます。これらは、スカラーデータ (文字列、キャラクタ、数字、等) か、バイナリデータ (画像、Protocol Buffer のシリアライズされた配列) の形になります。
 
-<img src="/img/collection_data_types.png" alt="Collection data types drawing | Momento Cache" width="80%"/>
+### 集合データ型 (CDT)
```
Comment (poppoerika) [inline i18n/ja/docusaurus-plugin-content-docs/current/develop/datatypes.md]: I think we can use `コレクション` here

File: i18n/ja/docusaurus-plugin-content-docs/current/develop/datatypes.md [hunk 3]
```diff
@@ -1,28 +1,31 @@
...
 
-#### Lists
-A list is a collection of ordered elements, sorted in the sequence each element was inserted.
+集合データ型は、関連するデータを一つの項目にグループ化したものです。それらはバイト配列の値として保存され、Momento SDK ではそれぞれに API が提供されています。例えば、DictionaryFetch、ListFetch、SetFetch 等です。
```
Comment (poppoerika) [inline i18n/ja/docusaurus-plugin-content-docs/current/develop/datatypes.md]: `集合データ型は`　-> `コレクションデータ型`

File: i18n/ja/docusaurus-plugin-content-docs/current/develop/datatypes.md [hunk 4]
```diff
@@ -1,28 +1,31 @@
...
+
+![集合データ型 | Momento Cache](/img/collection_data_types.png)
```
Comment (poppoerika) [inline i18n/ja/docusaurus-plugin-content-docs/current/develop/datatypes.md]: `[集合データ型`　-> `コレクションデータ型`

File: i18n/ja/docusaurus-plugin-content-docs/current/develop/datatypes.md [hunk 5]
```diff
@@ -39,12 +42,13 @@ A list is a collection of ordered elements, sorted in the sequence each element
 ]
 ```
 
-To learn the API methods, check the [API reference for list collection data types](./api-reference/list-collections.md).
+API メソッドは、[リスト集合データ型の API リファレンス](./api-reference/list-collections.md)をご覧下さい。
```
Comment (poppoerika) [inline i18n/ja/docusaurus-plugin-content-docs/current/develop/datatypes.md]: same comment as https://github.com/momentohq/public-dev-docs/pull/194/files#r1186332378

File: i18n/ja/docusaurus-plugin-content-docs/current/develop/datatypes.md [hunk 6]
```diff
@@ -54,10 +58,11 @@ An example of using a dictionary is when you need to store data together and ret
 }
 ```
 
-To learn the API methods, check the [API reference for dictionary collection data types](./api-reference/dictionary-collections.md).
+API メソッドは、[辞書集合データ型の API リファレンス](./api-reference/dictionary-collections.md)をご覧下さい。
```
Comment (poppoerika) [inline i18n/ja/docusaurus-plugin-content-docs/current/develop/datatypes.md]: same comment as https://github.com/momentohq/public-dev-docs/pull/194/files#r1186332378

File: i18n/ja/docusaurus-plugin-content-docs/current/develop/datatypes.md [hunk 7]
```diff
@@ -70,11 +75,11 @@ A set is an unordered collection of unique elements, each in string format. For
 }
 ```
 
-To learn the API methods, check the [API reference for set collection data types](./api-reference/set-collections.md).
+API メソッドは、[セット集合データ型の API リファレンス](./api-reference/set-collections.md)をご覧下さい。
```
Comment (poppoerika) [inline i18n/ja/docusaurus-plugin-content-docs/current/develop/datatypes.md]: same comment as https://github.com/momentohq/public-dev-docs/pull/194/files#r1186332378

File: i18n/ja/docusaurus-plugin-content-docs/current/develop/datatypes.md [hunk 8]
```diff
@@ -91,29 +96,29 @@ A sorted set is a collection of unique elements with a value (string) and score
 
 :::note
 
-If elements have the same score, they are sorted [lexicographically](https://www.dictionary.com/browse/lexicographically).
+もし要素が同じスコアの場合、[辞書的に](https://www.dictionary.com/browse/lexicographically)ソートされます。
 
 :::
 
-To learn the API methods, check the [API reference for sorted set collection data types](./api-reference/sorted-set-collections.md).
+API メソッドは [ソート済セット集合データ型の API リファレンス](./api-reference/sorted-set-collections.md)をご覧下さい。
```
Comment (poppoerika) [inline i18n/ja/docusaurus-plugin-content-docs/current/develop/datatypes.md]: same comment as https://github.com/momentohq/public-dev-docs/pull/194/files#r1186332378

File: i18n/ja/docusaurus-plugin-content-docs/current/develop/datatypes.md [hunk 9]
```diff
@@ -91,29 +96,29 @@ A sorted set is a collection of unique elements with a value (string) and score
...
-  <summary>How do I store a JSON document in Momento Cache?</summary>
-Use your favorite JSON library to serialize the JSON document into a byte array and insert that byte array into Momento Cache. You could also store each field value in your JSON document in a dictionary.
+  <summary>どうやれば Momento Cache に JSON ドキュメントを保存できますか？</summary>
+お好みの JSON ライブラリを使って、その JSON ドキュメントをバイト配列にシリアライズし、そのバイト配列を Momento Cache に挿入してください。または、JSON ドキュメントの各フィールドの値を辞書型に保存することもできます。
 </details>
 
 <details>
-  <summary>Does Momento Cache store nested data in collection data types?</summary>
-Not directly. Your best option is to store this data as a JSON object and then use your favorite JSON library to serialize the JSON document into a byte array and insert that byte array into Momento Cache.
+  <summary>Momento Cache は集合データ型でネストされたデータを保存できますか？</summary>
```
Comment (poppoerika) [inline i18n/ja/docusaurus-plugin-content-docs/current/develop/datatypes.md]: same comment as https://github.com/momentohq/public-dev-docs/pull/194/files#r1186332378

**Comment by riywo** [comment]
@poppoerika Addressed your comments! https://github.com/momentohq/public-dev-docs/compare/7e87d6b0413bd897499f73e7069aad6102645030..561e10dfe2a56341a7c641da5f04db1bf47c5617

**Review by poppoerika — APPROVED** [review]

**Pull request merged**